server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -193,139 +221,152 @@ private void verifyResponseReadFully(Header header, TransportResponseHandler<?>
-                                    + action
-                                    + "], available ["
-                                    + stream.available()
-                                    + "]; resetting"
-                            );
-                            assert ignoreDeserializationErrors : exception;
-                            throw exception;
-                        }
-                        final String executor = reg.getExecutor();
-                        if (ThreadPool.Names.SAME.equals(executor)) {
-                            try (var ignored = threadPool.getThreadContext().newTraceContext()) {
-                                try {
-                                    reg.processMessageReceived(request, transportChannel);
-                                } catch (Exception e) {
-                                    sendErrorResponse(reg.getAction(), transportChannel, e);
-                                }
-                            }
-                        } else {
-                            boolean success = false;
-                            request.incRef();
-                            try {
-                                threadPool.executor(executor)
-                                    .execute(threadPool.getThreadContext().preserveContextWithTracing(new AbstractRunnable() {
-                                        @Override
-                                        protected void doRun() throws Exception {
-                                            reg.processMessageReceived(request, transportChannel);
-                                        }
-
-                                        @Override
-                                        public boolean isForceExecution() {
-                                            return reg.isForceExecution();
-                                        }
-
-                                        @Override
-                                        public void onFailure(Exception e) {
-                                            sendErrorResponse(reg.getAction(), transportChannel, e);
-                                        }
-
-                                        @Override
-                                        public void onAfter() {
-                                            request.decRef();
-                                        }
-                                    }));
-                                success = true;
-                            } finally {
-                                if (success == false) {
-                                    request.decRef();
-                                }
-                            }
-                        }
-                    } finally {
-                        request.decRef();
-                    }
+    private static <T extends TransportRequest> void doHandleRequest(
+        RequestHandlerRegistry<T> reg,
+        T request,
+        TransportChannel transportChannel
```

Comment:
nit: rename to `channel` like with `handleRequestForking`, then the method signature fits on one line.